drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/projects_importer.rb

```diff
@@ -0,0 +1,222 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HmisExternalApis::AcHmis::Importers
+  class ProjectsImporter
+    include NotifierConfig
+
+    AbortImportException = Class.new(StandardError)
+
+    attr_accessor :attempt
+    attr_accessor :data_source
+    attr_accessor :dir
+
+    def initialize(dir:, key:, etag:)
+      self.attempt = ProjectsImportAttempt.where(etag: etag, key: key).first_or_initialize
+      self.data_source = HmisExternalApis::AcHmis.data_source
+      self.dir = dir
+    end
+
+    def run!
+      start
+      sanity_check
+      # validate
+      ProjectsImportAttempt.transaction do
+        upsert_funders
+        upsert_orgs
+        upsert_projects
+        upsert_walkins
+      end
+      analyze
+      finish
+    rescue AbortImportException => e
+      # FIXME: see if this is right
+      @notifier.ping(
+        'Failure in project importer',
+        {
+          exception: e,
+          info: nil,
+        },
+      )
+      Rails.logger.fatal e.message
+    end
+
+    def start
+      setup_notifier('HMIS Projects')
+      Rails.logger.info "Starting #{attempt.key}"
+      attempt.attempted_at = Time.now
+      attempt.status = ProjectsImportAttempt::STARTED
+      attempt.save!
+    end
+
+    def sanity_check
+      return unless Dir.glob("#{dir}/*csv").empty?
+
+      msg = "No csv files were found in #{attempt.key}"
+      Rails.logger.error(msg)
+      attempt.attempted_at = Time.now
+      attempt.status = ProjectsImportAttempt::FAILED
+      attempt.result = { error: msg }
+      attempt.save!
+      raise AbortImportException, msg
+    end
+
+    def validate
+      Rails.logger.info 'Validating CSVs (wip)'
+    end
+
+    def upsert_funders
+      generic_upsert(
+        file: 'Funder.csv',
+        conflict_target: ['"FunderID"', 'data_source_id'],
+        klass: GrdaWarehouse::Hud::Funder,
+      )
+    end
+
+    def upsert_orgs
+      generic_upsert(
+        file: 'Organization.csv',
+        conflict_target: ['"OrganizationID"', 'data_source_id'],
+        klass: GrdaWarehouse::Hud::Organization,
+      )
+    end
+
+    def upsert_projects
+      @project_result = generic_upsert(
+        file: 'Project.csv',
+        conflict_target: ['"ProjectID"', 'data_source_id'],
+        klass: GrdaWarehouse::Hud::Project,
+        ignore_columns: ['Walkin'],
+      )
+    end
+
+    # FIXME: I'm not sure how to execute this efficiently since I think we might have to fetch the user from the project.
+    # FIXME: What do we do if there's no user? I'm just skipping them for now.
```

Comment:
@eanders how do we usually handle the UserID field for imported data? I think in this case we can just set them all to the system user. `Hmis::Hud::User.system_user(hmis_data_source_id)`

Reply:
When I did that, I got a weird error about compound database keys. I could look again.

Reply:
They should be sending the UserID I'd think.  Maybe we need to request the User.csv file as well.

Reply:
Got this working. Just needed to not include the user in the where clause.